src/plans/scaled_objective.jl

```diff
@@ -0,0 +1,136 @@
+@doc """
+    ScaledManifoldObjective{E, O2, O1<:AbstractManifoldObjective{E},F} <:
+       AbstractDecoratedManifoldObjective{E,O2}
+
+Declare an objective to be defined as a scaled version of an existing objective.
+
+This rescales all involved functions.
+
+For now the functions rescaled are
+
+* the cost
+* the gradient
+* the Hessian
+
+# Fields
+
+* `objective`: the objective that is defined in the embedding
+* `scale=1`: the scaling applied
+
+# Constructors
+
+    ScaledManifoldObjective(objective, scale=1)
```

Comment:
```suggestion
    ScaledManifoldObjective(objective, scale::Real=1)
```
I'd suggest restricting it to real numbers.